### PR TITLE
Create config at compile time

### PR DESCRIPTION
**Pull request by kklin**
Once this is merged, we should disable dockerhub autobuilds!

### CODE REVIEW

**Comment by kklin** [comment]
@ejj please merge when you're free

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +0,0 @@
-From haproxy:1.6.4
```
Comment (mmussomele) [inline Dockerfile]: Should the commit message read "can create it entirely at compile time"?

Comment (kklin) [inline Dockerfile]: Yes, thanks @mmussomele!

**Comment by ejj** [comment]
What's the point of the first patch, given that the second deletes everything?

**Comment by LuiseV** [comment]
This is what happens when I just look at the final diff. Lesson learned!

**Comment by kklin** [comment]
The two commits achieve completely separate goals -- if it ends up that we actually _do_ need to block on DNS resolution it'd be easier to pinpoint the commit that broke the spec/revert it.

**Pull request merged**